apis/v1beta1/object_reference_types.go

```diff
@@ -99,9 +99,33 @@ type BackendObjectReference struct {
 	// +kubebuilder:default=""
 	Group *Group `json:"group,omitempty"`
 
-	// Kind is kind of the referent. For example "HTTPRoute" or "Service".
+	// Kind is the Kubernetes resource kind of the referent. For example
+	// "Service" or "HTTPRoute".
+	//
 	// Defaults to "Service" when not specified.
 	//
+	// Selectors in Kubernetes Services provide an abstraction around Pods,
+	// meaning that when a Service uses selectors Gateway API implementations
+	// can reason about what Namespaces the traffic will go to and can therefore
+	// enforce namespace boundaries (e.g. ReferenceGrant). As such we consider
+	// Services that utilize selectors to be considered supported when used as
```

Comment:
Commented below, but I don't think this is the right thing to make the distinction on.